COMMIT_GUIDELINES.md

```diff
@@ -0,0 +1,141 @@
+
+Message Body
+
+Footer
+```
+
+The blank lines in between are mandatory. A commit **must** include a `TYPE` and a `Summary` and **may**
+additionally contain any of the other components listed here.
+
+
+## Subject line
+
+The first line ("Subject line") should not exceed 50-70 characters. This is what gets displayed on GitHub at first glance, so it
+should contain the most important information. In order for it to be as short and precise as possible, there is the `TYPE` and
+optionally a `Scope`. With these it should already be clear what this commit is about in general. The short summary should
+then include further information that is important to understand the general idea of this commit at a glance.
+
+
+### TYPE
+
+The `TYPE` is one of the following:
+
+| -------- | --------------- | ----------- |
+| **TYPE** | **Description** | **Example** |
+| -------- | --------------- | ----------- |
+| BREAK    | A breaking change - not backwards-compatible. | A change in protocol (e.g. change UDP message format) |
+| FEAT     | Introduction of a new feature (or extension of an existing one) | |
+| FIX      | A bug fix | |
+| FORMAT   | Change of formatting - does not influence how the code works | Change indentation of a line; Add braces around `if` body |
+| DOCS     | Changes to the documentation (either in-source or out-of-source) | Add a Doxygen comment to a function |
+| TEST     | Adds, changes or removes a test-case | |
+| MAINT    | Maintenance - Change of non-code files | Change of the README |
+| CI       | Changed something for the CI (continous integration) | Update TravisCI to use newer ubuntu version |
+| REFAC   | Code refactoring | Renamed variable `x` to `y` |
+
+The `TYPE` has to be in **all-uppercase** in order for it to stand out.
+
+If you feel like you need to use 2 or more types for a single commit but *can't split it* into multiple commits, you can
+combine types with `/`: `FEAT/CI: <Summary>`
+
+
+### Scope
+
+What area is the change about. For now we don't have fixed scope keywords. A scope could be something like `ui`, `client`,
+`server`, `ice`, `grpc`, ...
+
+
+### Summary
+
+The summary is the heart of the subject line. It should contain a **very brief** summary of what you did in that commit.
+In order to make this as short as possible, you may use grammatically incorrect sentences
+("Add ability" instead of "add the ability")
+
+In general the summary should answer the question "Applying this commit will ..." where "..." is your summary.
+
+If your summary contains "and", you should probably split your commit up.
+
+Note: Issue references (such as #2305) **must not** be used in the summary!
+
+
+## Message Body
+
+Here you give more details about the commit. Why is it necessary and what are the details of the change. You can use
+multiple paragraphs for this and be as verbose as you want.
+
+The message body should reference issues that are related to this change, but also provide a short summary of what that
```

Comment:
I just read this and a question popped into my head: "Should I also link to the issue I'm referencing?".

Maybe for others this is true as well. Should one link to the issue, or does writing the `#id` suffice? I would like it more if this was mentioned in the guide, so that I could be at ease and not link if I don't need to, or link right away and not mess it up :)

Reply:
Good point. #id is enough as this is unique. The link is either automatically generated by GitHub or can be found rather trivially by looking up GitHub link structures :+1: